addons/mail/static/src/messaging/component/notification_list/notification_list.js

```diff
@@ -53,49 +50,47 @@ class NotificationList extends Component {
      * @private
      */
     async _loadPreviews() {
-        this.storeDispatch('loadThreadPreviews',
-            this.notifications
-                .filter(notification => notification.threadLocalId)
-                .map(notification => notification.threadLocalId)
-        );
+        const { notifications } = this.storeProps;
+        const threads = notifications
+            .filter(notification => notification.thread)
+            .map(notification => this.env.entities.Thread.get(notification.thread));
+        this.env.entities.Thread.loadPreviews(threads);
     }
 
     /**
      * @private
-     * @param {Object} state
      * @param {Object} props
      */
-    _useStoreSelector(state, props) {
-        const threads = this._useStoreSelectorThreads(state, props);
+    _useStoreSelector(props) {
+        const threads = this._useStoreSelectorThreads(props);
         const notifications = threads.map(thread => {
             return {
-                threadLocalId: thread.localId,
+                thread: thread,
                 type: 'thread',
                 uniqueId: thread.localId,
             };
         });
         return {
-            isMobile: state.isMobile,
+            isDeviceMobile: this.env.messaging.device.isMobile,
             notifications,
         };
     }
 
     /**
      * @private
-     * @param {Object} state
      * @param {Object} props
-     * @returns {Object[]} threads
+     * @returns {mail.messaging.entity.Thread[]}
```

Comment:
throw not documented, though really it's more like a programmer error than an actual signature of the method